DTML.EduBot/Dialogs/BotQuestionsDialog.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace DTML.EduBot.Dialogs
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Bot.Builder.Dialogs;
+    using Microsoft.Bot.Builder.Luis;
+    using Microsoft.Bot.Builder.Luis.Models;
+    using DTML.EduBot.Common;
+    using DTML.EduBot.Qna;
+    using Microsoft.Azure;
+    using Attributes;
+    using System.Linq;
+    using DTML.EduBot.Constants;
+
+    [PreConfiguredLuisModel]
+    [PreConfiguredQnaModel]
+    [Serializable]
+    public partial class BotQuestionsDialog : QnaLuisDialog<object>
+    {
+        [LuisIntent("BotQuestions")]
+        public async Task HandleBotQuestionsIntent(IDialogContext context, LuisResult result)
+        {
+            if (result.Entities.Any(e => e.Type == BotEntities.Name))
+            {
+                await context.PostAsync("My name is " + BotPersonality.BotName);
```

Comment:
Any message that we send to a user should be a constant. You can change it in this PR or create a separate PR for this. But we need to do this ASAP.